src/go/k8s/controllers/redpanda/cluster_controller_configuration_test.go

```diff
@@ -895,3 +891,13 @@ func clusterConfiguredConditionStatusGetter(key client.ObjectKey) func() bool {
 		return cond != nil && cond.Status == corev1.ConditionTrue
 	}
 }
+
+func clusterUpdater(cluster *v1alpha1.Cluster, upd func()) func() error {
+	return func() error {
+		if err := k8sClient.Get(context.Background(), client.ObjectKeyFromObject(cluster), cluster); err != nil {
+			return err
+		}
+		upd()
+		return k8sClient.Update(context.Background(), cluster)
+	}
+}
```

Comment:
NIT: Maybe it would be better to provide only the NamespacedName? We don't use the cluster to assert anything. There is even more explicit update and get pattern:

https://github.com/redpanda-data/redpanda/blob/a98c82b3d17b7fc058d4ea831dcd427be6865047/src/go/k8s/controllers/redpanda/cluster_controller_configuration_test.go#L660-L675

```suggestion
func clusterUpdater(clusterNamespacedName types.NamespacedName, upd func(*v1alpha1.Cluster)) func() error {
	return func() error {
		cl := &v1alpha1.Cluster{}
		if err := k8sClient.Get(context.Background(), clusterNamespacedName, cl); err != nil {
			return err
		}
		upd(cl)
		return k8sClient.Update(context.Background(), cl)
	}
}
```